src/vs/workbench/contrib/notebook/browser/viewModel/notebookOutlineProvider.ts

```diff
@@ -53,29 +57,19 @@ export class NotebookCellOutlineProvider {
 		@IConfigurationService private readonly _configurationService: IConfigurationService,
 	) {
 		this._outlineEntryFactory = new NotebookOutlineEntryFactory(notebookExecutionStateService);
-		const delayerRecomputeActive = this._disposables.add(new Delayer(10));
-		this.delayRecomputeActive = () => delayerRecomputeActive.trigger(() => this._recomputeActive());
-
-		this._disposables.add(Event.debounce<void, void>(
-			_editor.onDidChangeSelection,
-			(last, _current) => last,
-			200
-		)(() => {
-			this.delayRecomputeActive();
+		const delayerRecomputeActive = this._disposables.add(new Delayer(200));
+
+		this._disposables.add(_editor.onDidChangeSelection(() => {
+			delayerRecomputeActive.trigger(() => this._recomputeActive());
 		}, this))
-		this._disposables.add(Event.debounce<INotebookViewCellsUpdateEvent, INotebookViewCellsUpdateEvent>(
```

Comment:
This is the same as existing event handler looking at the notebook content changes.